engine/src/main/java/org/terasology/rendering/nui/layers/ingame/metrics/DebugOverlay.java

```diff
@@ -168,6 +168,11 @@ public Boolean get() {
     @Override
     public void update(float delta) {
         if (metricsLabel != null) {
+            if (metricsModes.isEmpty()) {
+                metricsLabel.setText("No metric modes are available!");
+                return;
+            }
+
             metricsLabel.setText(metricsModes.get(currentMode).getMetrics());
```

Comment:
I guess my concern here might be a life-cycle issue. Is a DebugOverlay instance and its labels alive at all times? Ideally it shouldn't be, only when the user presses F3 it should come alive. And are its labels instantiated only when there is content to put into them?